packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -204,6 +204,7 @@ export class Bundling implements cdk.BundlingOptions {
       ...this.props.banner ? [`--banner:js=${JSON.stringify(this.props.banner)}`] : [],
       ...this.props.footer ? [`--footer:js=${JSON.stringify(this.props.footer)}`] : [],
       ...this.props.charset ? [`--charset=${this.props.charset}`] : [],
+      ...this.props.esbuildArgs ? [Object.keys(this.props.esbuildArgs).map((key) => `${key}=${this.props.esbuildArgs![key]}`).join(' ')] : [],
```

Comment:
```suggestion
      ...this.props.esbuildArgs ? [Object.entries(this.props.esbuildArgs).map(([key, value]) => `${key}=${value}`).join(' ')] : [],
```


\+ should `value` be enclosed with double quotes? `${key}="${value}"` to cover args where value is a path that can contain whitespaces? or should the user take care of this?

Reply:
Ah and what about args that don't have values like `--keep-names`?

Reply:
Good suggestions, I'll get those implemented. Thanks!

Reply:
> Ah and what about args that don't have values like --keep-names?

@jogold because the type for `esbuildArgs` is an object we can always expect a value being assigned to each key. I think it's reasonable to assume that the user will need to pass a truthy value into the value of those boolean args. 

i.g.

```ts
...
esbuildArgs: {
      '--log-limit': '0',
      '--resolve-extensions': '.ts,.js',
      '--keep-names': true,
    },
```

I'll add this into the README example and tests for clarification.

Reply:
But `esbuild` doesn't accept this syntax in the CLI...

```
> error: Invalid build flag: "--splitting=true"
```

You need to update the logic to treat the Booleans differently (output only the key)

Reply:
I'm not able to reproduce this error. See my tests below:

```bash
$ esbuild src/* --bundle --outdir=output/splitting-novalue --format=esm --splitting

  output/splitting-novalue/app.js             133.9kb
  output/splitting-novalue/chunk-BJ7SH3MN.js  131.7kb
  output/splitting-novalue/chunk-IZI55ZEC.js    447b 
  output/splitting-novalue/chunk-QND2VRQK.js    442b 
  output/splitting-novalue/GreenSpinner.js      138b 
  output/splitting-novalue/BlueSpinner.js       136b 
  output/splitting-novalue/Spinner.js            98b 

$ esbuild src/* --bundle --outdir=output/splitting-withvalue-true --format=esm --splitting=true

  output/splitting-withvalue-true/app.js             133.9kb
  output/splitting-withvalue-true/chunk-BJ7SH3MN.js  131.7kb
  output/splitting-withvalue-true/chunk-IZI55ZEC.js    447b 
  output/splitting-withvalue-true/chunk-QND2VRQK.js    442b 
  output/splitting-withvalue-true/GreenSpinner.js      138b 
  output/splitting-withvalue-true/BlueSpinner.js       136b 
  output/splitting-withvalue-true/Spinner.js            98b 

$ esbuild src/* --bundle --outdir=output/splitting-withvalue-false --format=esm --splitting=false

  output/splitting-withvalue-false/app.js           266.0kb
  output/splitting-withvalue-false/GreenSpinner.js  132.0kb
  output/splitting-withvalue-false/BlueSpinner.js   132.0kb
  output/splitting-withvalue-false/Spinner.js       131.7kb

$ esbuild src/* --bundle --outdir=output/splitting-withvalue-wrapped-true --format=esm --splitting="true"

  output/splitting-withvalue-wrapped-true/app.js             133.9kb
  output/splitting-withvalue-wrapped-true/chunk-BJ7SH3MN.js  131.7kb
  output/splitting-withvalue-wrapped-true/chunk-IZI55ZEC.js    447b 
  output/splitting-withvalue-wrapped-true/chunk-QND2VRQK.js    442b 
  output/splitting-withvalue-wrapped-true/GreenSpinner.js      138b 
  output/splitting-withvalue-wrapped-true/BlueSpinner.js       136b 
  output/splitting-withvalue-wrapped-true/Spinner.js            98b 

$ esbuild src/* --bundle --outdir=output/splitting-withvalue-wrapped-false --format=esm --splitting="false"

  output/splitting-withvalue-wrapped-false/app.js           266.0kb
  output/splitting-withvalue-wrapped-false/GreenSpinner.js  132.0kb
  output/splitting-withvalue-wrapped-false/BlueSpinner.js   132.0kb
  output/splitting-withvalue-wrapped-false/Spinner.js       131.7kb

✨  Done in 1.04s.
```

Could you share the command you're using to get that "Invalid build flag" error?

Reply:
OK it works with `esbuild` >= 0.14.24 published 8 days ago.

You can try with `npx esbuild@0.14.23 ...`.

Reply:
I suggest this https://github.com/aws/aws-cdk/pull/19343